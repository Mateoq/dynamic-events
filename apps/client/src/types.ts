export enum Direction {
  LEFT = 'left',
  RIGHT = 'right'
}

export enum CalendarType {
  MOTH = 'MONTH',
  WEEK = 'WEEK',
  DAY = 'DAY'
}

export enum ToastType {
  SUCCESS = 'success',
  ERROR = 'error'
}

export type Time = {
  key: string;
  hour: number;
  minutes: string[];
};

export type Day = {
  key: string;
  value: number;
  isToday: boolean;
  times: Record<string, Time>;
};

export type Week = {
  key: string;
  value: number;
  firstIndex: number;
  lastIndex: number;
  days: Record<string, Day>;
};

export type Month = {
  key: string;
  value: number;
  firstIndex: number;
  lastIndex: number;
  weeks: Record<string, Week>;
};

export type Year = {
  key: string;
  value: number;
  months: Record<string, Month>;
};

export type CalendarData = {
  years: Record<string, Year>;
};

export type DayToRender = {
  data: Day;
  week: number;
};

export type CalendarState = {
  year: number;
  month: number;
  week: number;
  day: number;
};

export type Event = {
  id: string;
  title: string;
  description: string;
  city: string;
  startYear: number;
  startMonth: number;
  startDate: number;
  endYear: number;
  endMonth: number;
  endDate: number;
  time: number;
  fullDay: boolean;
  color: string;
};

export type CreateEventDto = {
  title: string;
  description: string;
  city: string;
  startYear: number;
  startMonth: number;
  startDate: number;
  endYear: number;
  endMonth: number;
  endDate: number;
  time: number;
  fullDay: boolean;
  userId: string;
};

export type UpdateEventDto = Partial<CreateEventDto>;

export type CreateEventApiDto = CreateEventDto & {
  color: string;
};

export enum TransactionResultType {
  SUCCESS = 'TRAN_SUCCESS',
  EMAIL_USED = 'TRAN_EMAIL_USED',
  WRONG_ID = 'TRAN_WRONG_ID',
  WRONG_USER = 'TRAN_WRONG_USER',
  ERROR = 'TRAN_ERROR',
  UNKNOWN_ERROR = 'TRAN_UNKNOWN_ERROR'
}

export interface TransactionError {
  message: string;
}

export type TransactionResult<T> = {
  status: boolean;
  type: TransactionResultType;
  data: T;
  error: TransactionError | null;
};

export type EventMap = Record<string, Event>;

export type TimeEventMap = Record<number, string[]>;

export type DayEventMap = Record<number, TimeEventMap>;

export type MonthEventMap = Record<number, DayEventMap>;

export type YearEventMap = Record<number, MonthEventMap>;

export type UserEventMap = YearEventMap;

export type EventData = {
  events: EventMap;
  calendarEvents: UserEventMap;
};
// --------------------------------

export type DateItem = {
  key: string;
  date: Date;
  isToday: boolean;
};

export type DateMap = {
  [key: string]: DateItem;
};

export type MonthData = {
  firstIndex: number;
  lastIndex: number;
  dates: DateMap;
};

export type MonthMap = {
  [key: string]: MonthData;
};

export type CalendarMonthData = {
  currentKey: string;
  months: MonthMap;
  currentDate: {
    year: number;
    month: number;
  };
};
