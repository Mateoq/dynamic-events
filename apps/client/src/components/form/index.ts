export * from './Button';
export * from './SelectInput';
export * from './TextInput';
export * from './DatePicker';
export * from './CheckBox';
