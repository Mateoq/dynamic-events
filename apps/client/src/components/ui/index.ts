export * from './Calendar';
export * from './LoadingIcon';
export * from './Modal';
export * from './Toast';
export * from './EventCard';
