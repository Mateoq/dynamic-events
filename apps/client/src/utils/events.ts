import { EventMap, Event, UserEventMap } from '@/types';
import { getLastMonthDate } from './calendar';
import { MAX_CALENDAR_MONTH } from '@/constants';

export function mapEvents(events: Event[]): EventMap {
  const map: EventMap = {};

  for (const event of events) {
    map[event.id] = event;
  }

  return map;
}

export function mapCalendarEvents(events: Event[]): UserEventMap {
  const map: UserEventMap = {};

  for (const event of events) {
    const { startYear, startMonth, startDate, endYear, endMonth, endDate } =
      event;

    for (let year = startYear; year <= endYear; year++) {
      const firstMonth = year === startYear ? startMonth : 0;
      const lastMonth = year === endYear ? endMonth : MAX_CALENDAR_MONTH - 1;
      if (!map[year]) {
        map[year] = {};
      }

      console.log('YEAR', map);

      for (let month = firstMonth; month <= lastMonth; month++) {
        const firstDay =
          year === startYear && month === startMonth ? startDate : 1;
        const lastDay =
          year === endYear && month === endMonth
            ? endDate
            : getLastMonthDate(year, month);

        if (!map[year][month]) {
          map[year][month] = {};
        }

        console.log('MONTH', map, firstDay, lastDay, event);

        for (let day = firstDay; day <= lastDay; day++) {
          if (!map[year][month][day]) {
            map[year][month][day] = {};
          }

          if (!map[year][month][day][event.time]) {
            map[year][month][day][event.time] = [];
          }

          console.log('TIME', map);

          map[year][month][day][event.time].push(event.id);
        }
      }
    }
  }

  return map;
}
