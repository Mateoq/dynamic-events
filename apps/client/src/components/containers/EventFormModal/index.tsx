'use client';

import React from 'react';
import { Controller } from 'react-hook-form';
// import { DateValueType } from 'react-tailwindcss-datepicker';

import { Modal } from '@/components/ui';
import {
  TextInput,
  SelectInput,
  SelectItem,
  DatePicker,
  CheckBox,
  Button
} from '@/components/form';
import { useEventForm } from '@/hooks/useEventForm';
import { useEventModalStore } from '@/stores/event-modal-store';
import { CITIES, HOURS } from '@/constants';

const citiesOptions: SelectItem[] = CITIES.map((city) => ({
  label: city,
  value: city
}));

const hourOptions: SelectItem[] = HOURS.map((hour, index) => ({
  label: hour,
  value: index.toString()
}));

export const EventFormModal: React.FC = () => {
  const { errors, control, reset, register, watch, isLoading, onSubmit } =
    useEventForm();
  const { isOpen, hideModal } = useEventModalStore();

  const handleCloseModal = () => {
    reset();
    hideModal();
  };

  const isChecked = watch('fullDay');

  return (
    <Modal isOpen={isOpen} onClose={handleCloseModal} title="Event Form">
      <form onSubmit={onSubmit} className="flex flex-col gap-y-4">
        <TextInput
          {...register('title')}
          label="Title"
          message="Event title require"
          error={!!errors.title}
        />
        <TextInput
          {...register('description')}
          label="Description"
          large
          message="Event description required"
          error={!!errors.description}
        />
        <SelectInput
          {...register('city')}
          label="City"
          options={citiesOptions}
          message="Event city required"
          error={!!errors.city}
        />
        <Controller
          control={control}
          name="startDate"
          render={({ field }) => (
            <DatePicker
              label="Start Date"
              selected={field.value}
              onChange={(value) => field.onChange(value)}
              error={!!errors.startDate}
              message="Starting date required"
            />
          )}
        />
        <Controller
          control={control}
          name="endDate"
          render={({ field }) => (
            <DatePicker
              label="End Date"
              selected={field.value}
              onChange={(value) => field.onChange(value)}
              error={!!errors.endDate}
              message="End date required"
            />
          )}
        />
        <SelectInput
          {...register('time')}
          label="Time"
          options={hourOptions}
          message="Event time required"
          error={!!errors.time}
        />
        <CheckBox
          {...register('fullDay')}
          label="Full Day"
          isChecked={isChecked}
        />
        <div>
          <Button type="submit" isLoading={isLoading}>
            Save
          </Button>
        </div>
      </form>
    </Modal>
  );
};
