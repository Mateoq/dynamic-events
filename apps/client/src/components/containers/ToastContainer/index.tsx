'use client';

import React, { useEffect } from 'react';

import { Toast } from '@/components/ui';
import { useToastStore } from '@/stores/toast-store';
import { sleep } from '@/utils/functions';
import { TOAST_SLEEP_TIME } from '@/constants';

export const ToastContainer: React.FC = () => {
  const { isOpen, text, type, hideToast } = useToastStore();

  const sleepNHide = async () => {
    await sleep(TOAST_SLEEP_TIME);
    hideToast();
  };

  useEffect(() => {
    if (isOpen) {
      sleepNHide();
    }
  }, [isOpen]);

  return <Toast isOpen={isOpen} text={text} type={type} />;
};
