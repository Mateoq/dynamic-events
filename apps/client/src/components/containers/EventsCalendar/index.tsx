'use client';

import React, { useState, useEffect } from 'react';

import { Providers } from '../providers';
import { Calendar } from '@/components/ui';
import { ToastContainer } from '../ToastContainer';
import { EventFormModal } from '../EventFormModal';
import { useCalendarStateStore } from '@/stores/calendar-state-store';
import { useEventModalStore } from '@/stores/event-modal-store';
import { useUserStore } from '@/stores/user-store';
import { CalendarData, CalendarType, CalendarState, Direction } from '@/types';
import {
  createYearData,
  computeMonthState,
  computeWeekState,
  computeDayState
} from '@/utils/calendar';

export type EventsCalendarProps = {
  initialData: CalendarData;
  initialState: CalendarState;
};

export const EventsCalendar: React.FC<EventsCalendarProps> = (props) => {
  const { initialData, initialState } = props;
  const { calendarState, setState: setCalendarState } = useCalendarStateStore();
  const showModal = useEventModalStore((state) => state.showModal);
  const [calendarData, setCalendarData] = useState<CalendarData>(initialData);
  const [currentView, setCurrentView] = useState(CalendarType.MOTH);

  const setUserData = useUserStore((state) => state.setUserData);

  const handleNavigation = (direction: Direction) => {
    console.log('NAV', direction);
    let nextState = { ...calendarState };
    switch (currentView) {
      case CalendarType.MOTH:
        nextState = computeMonthState(direction, nextState, calendarData);
        break;
      case CalendarType.WEEK:
        nextState = computeWeekState(direction, nextState, calendarData);
        break;
      case CalendarType.DAY:
        nextState = computeDayState(direction, nextState, calendarData);
        break;
    }

    if (!calendarData.years[nextState.year]) {
      setCalendarData((prev) => ({
        ...prev,
        years: {
          ...prev.years,
          [nextState.year]: createYearData(nextState.year)
        }
      }));
    }

    setCalendarState(nextState);
  };

  const openEventModal = () => {
    console.log('OPEN_MODAL');
    showModal();
  };

  const handleChangeView = (type: CalendarType) => {
    setCurrentView(type);
  };

  const openEventOnModal = (eventId: string) => {
    showModal(eventId);
  };

  useEffect(() => {
    setUserData('d7ceb2c2-e413-4353-bae4-c8553db1614c', 'test@example.com');
    setCalendarState(initialState);
  }, []);

  return (
    <Providers>
      <>
        <section className="px-24 py-20">
          <Calendar
            data={calendarData}
            type={currentView}
            onNav={handleNavigation}
            onCreateEvent={openEventModal}
            onChangeView={handleChangeView}
            onClickEvent={openEventOnModal}
          />
        </section>
        <ToastContainer />
        <EventFormModal />
      </>
    </Providers>
  );
};
