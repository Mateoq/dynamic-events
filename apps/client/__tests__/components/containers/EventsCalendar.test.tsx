import '@testing-library/jest-dom';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { EventsCalendar } from '@/components/containers';
import { CalendarType } from '@/types';
import {
  createInitialCalendarState,
  initialCalendarData,
  getEvents,
  emptySuccessTransactionResult
} from '../../../test-mocks';
import { sleep } from '@/utils/functions';

const state = createInitialCalendarState();

jest.mock('@/hooks/useEventData', () => ({
  useEventData: () => ({
    data: getEvents(),
    isLoading: false,
    refetchData: jest.fn()
  })
}));

const mockedFn = jest.fn();
jest.mock('@/hooks/useMutateEvent', () => ({
  useMutateEvent: () => ({
    mutate: () =>
      new Promise((resolve) => {
        mockedFn();
        resolve(emptySuccessTransactionResult);
      }),
    isLoading: false
  })
}));

describe('EventsCalendar', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', async () => {
    const { container } = render(
      <EventsCalendar initialData={initialCalendarData} initialState={state} />
    );
    const title = await screen.findByText('October - 2024');

    expect(title).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render month view by default', async () => {
    render(
      <EventsCalendar initialData={initialCalendarData} initialState={state} />
    );
    const currentStateDay = await screen.findByText('17');

    expect(currentStateDay).toBeInTheDocument();
  });

  it('should change to week view', async () => {
    render(
      <EventsCalendar initialData={initialCalendarData} initialState={state} />
    );
    const viewSelect = await screen.findByRole('listbox');
    await userEvent.selectOptions(viewSelect, CalendarType.WEEK);

    const time = await screen.findAllByText('1:00');

    expect(time[0]).toBeInTheDocument();
  });

  it('should change to day view', async () => {
    render(
      <EventsCalendar initialData={initialCalendarData} initialState={state} />
    );
    const viewSelect = await screen.findByRole('listbox');
    await userEvent.selectOptions(viewSelect, CalendarType.DAY);

    const time = await screen.findAllByText('1:00');

    expect(time[0]).toBeInTheDocument();
    expect(time).toHaveLength(1);
  });

  it('should open event form modal', async () => {
    render(
      <EventsCalendar initialData={initialCalendarData} initialState={state} />
    );

    const newEventBtn = await screen.findByText('New Event');

    await userEvent.click(newEventBtn);

    const titleLabel = await screen.findByText('Title');

    expect(titleLabel).toBeInTheDocument();
  });

  it('should close event form modal', async () => {
    render(
      <EventsCalendar initialData={initialCalendarData} initialState={state} />
    );

    const newEventBtn = await screen.findByText('New Event');

    await userEvent.click(newEventBtn);

    const closeBtn = await screen.findByLabelText('Close Modal');

    await userEvent.click(closeBtn);

    // Wait until modal fades out
    await sleep(1000);

    const titleInput = screen.queryByLabelText('Title');

    expect(titleInput).not.toBeInTheDocument();
  });

  it('should go to previous month', async () => {
    render(
      <EventsCalendar initialData={initialCalendarData} initialState={state} />
    );
    const prevBtn = await screen.findByLabelText('Previous');

    await userEvent.click(prevBtn);

    const monthTitle = await screen.findByText('September - 2024');

    expect(monthTitle).toBeInTheDocument();
  });

  it('should go to next month', async () => {
    render(
      <EventsCalendar initialData={initialCalendarData} initialState={state} />
    );
    const nextBtn = await screen.findByLabelText('Next');

    await userEvent.click(nextBtn);

    const monthTitle = await screen.findByText('November - 2024');

    expect(monthTitle).toBeInTheDocument();
  });

  it('should open event form modal when clicking event card', async () => {
    render(
      <EventsCalendar initialData={initialCalendarData} initialState={state} />
    );
    const eventCards = await screen.findAllByText('ANother');

    await userEvent.click(eventCards[0]);

    const titleInput = await screen.findByLabelText('Title');

    expect(titleInput).toBeInTheDocument();
    expect(titleInput).toHaveValue('ANother');
  });

  it('should submit to create an event', async () => {
    render(
      <EventsCalendar initialData={initialCalendarData} initialState={state} />
    );

    const newEventBtn = await screen.findByText('New Event');

    await userEvent.click(newEventBtn);

    const titleInput = await screen.findByLabelText('Title');
    const descriptionInput = await screen.findByLabelText('Description');
    const cityInput = await screen.findByLabelText('City');
    const startDateInput = await screen.findByLabelText('Start Date');
    const endDateInput = await screen.findByLabelText('End Date');
    const timeInput = await screen.findByLabelText('Time');

    await userEvent.type(titleInput, 'title');
    await userEvent.type(descriptionInput, 'description');
    await userEvent.selectOptions(cityInput, 'New York');

    // Change start date DatePicker value
    await userEvent.click(startDateInput);
    const startOptions = await screen.findAllByRole('option');
    await userEvent.click(startOptions[1]);
    await userEvent.click(titleInput);

    // Change end date DatePicker value
    await userEvent.click(endDateInput);
    const endOptions = await screen.findAllByRole('option');
    await userEvent.click(endOptions[2]);
    await userEvent.click(titleInput);

    await userEvent.selectOptions(timeInput, '12');

    const submitBtn = await screen.findByRole('button', { name: 'Save' });

    fireEvent.submit(submitBtn);

    await waitFor(() => {
      expect(mockedFn).toHaveBeenCalled();
      expect(titleInput).not.toBeInTheDocument();
    });
  });
});
