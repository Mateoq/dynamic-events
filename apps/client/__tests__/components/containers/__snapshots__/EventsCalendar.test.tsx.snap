// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsCalendar should match snapshot 1`] = `
<div>
  <section
    class="px-24 py-20"
  >
    <div
      class="w-full"
    >
      <div
        class="w-full flex flex-col"
      >
        <div
          class="w-full flex justify-between bg-gray-100 px-2 py-3"
        >
          <div
            class="flex gap-x-2"
          >
            <button
              aria-label="Previous"
              class="transition-opacity hover:opacity-70 active:opacity-100"
            >
              <svg
                fill="currentColor"
                height="30px"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 320 512"
                width="30px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256 246.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z"
                />
              </svg>
            </button>
            <button
              aria-label="Next"
              class="transition-opacity hover:opacity-70 active:opacity-100"
            >
              <svg
                fill="currentColor"
                height="30px"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 320 512"
                width="30px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
                />
              </svg>
            </button>
          </div>
          <div
            class="flex flex-1 items-center justify-center text-center"
          >
            <p
              class="text-xl font-semibold text-gray-900"
            >
              October
               - 
              2024
            </p>
          </div>
          <div
            class="flex gap-x-2 items-center"
          >
            <label
              class="flex flex-col w-fit"
            >
              <div
                class="relative w-fit cursor-pointer"
              >
                <select
                  class="min-h-11 border-2 px-2.5 rounded appearance-none cursor-pointer border-gray-700 min-w-64"
                  role="listbox"
                >
                  <option
                    value="MONTH"
                  >
                    Month
                  </option>
                  <option
                    value="WEEK"
                  >
                    Week
                  </option>
                  <option
                    value="DAY"
                  >
                    Day
                  </option>
                </select>
                <div
                  class="h-full w-10 absolute right-0 top-0 flex items-center justify-center"
                >
                  <svg
                    fill="currentColor"
                    height="18px"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="18px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                    />
                  </svg>
                </div>
              </div>
            </label>
            <button
              class="min-w-24 h-10 px-3o flex items-center justify-center bg-gray-950 text-white text-lg hover:bg-gray-800 active:bg-gray-700 transition-colors"
            >
              New Event
            </button>
          </div>
        </div>
        <div
          class="grid grid-cols-7 grid-rows-calendar bg-gray-400 gap-1"
        >
          <div
            class="bg-gray-400 flex items-center justify-center"
          >
            <span
              class="text-gray-900 text-xl text-center"
            >
              Sunday
            </span>
          </div>
          <div
            class="bg-gray-400 flex items-center justify-center"
          >
            <span
              class="text-gray-900 text-xl text-center"
            >
              Monday
            </span>
          </div>
          <div
            class="bg-gray-400 flex items-center justify-center"
          >
            <span
              class="text-gray-900 text-xl text-center"
            >
              Tuesday
            </span>
          </div>
          <div
            class="bg-gray-400 flex items-center justify-center"
          >
            <span
              class="text-gray-900 text-xl text-center"
            >
              Wednesday
            </span>
          </div>
          <div
            class="bg-gray-400 flex items-center justify-center"
          >
            <span
              class="text-gray-900 text-xl text-center"
            >
              Thursday
            </span>
          </div>
          <div
            class="bg-gray-400 flex items-center justify-center"
          >
            <span
              class="text-gray-900 text-xl text-center"
            >
              Friday
            </span>
          </div>
          <div
            class="bg-gray-400 flex items-center justify-center"
          >
            <span
              class="text-gray-900 text-xl text-center"
            >
              Saturday
            </span>
          </div>
          <div
            class="h-full bg-gray-300"
          />
          <div
            class="h-full bg-gray-300"
          />
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              01
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              02
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              03
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              04
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              05
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              06
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              07
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              08
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              09
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              10
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              11
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              12
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              13
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              14
            </span>
            <div
              class="flex flex-col w-full"
            >
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(198, 205, 101);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  Test
                </p>
              </button>
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(84, 59, 196);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  Test 2
                </p>
              </button>
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(76, 148, 200);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  ANother
                </p>
              </button>
            </div>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              15
            </span>
            <div
              class="flex flex-col w-full"
            >
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(198, 205, 101);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  Test
                </p>
              </button>
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(76, 148, 200);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  ANother
                </p>
              </button>
            </div>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              16
            </span>
            <div
              class="flex flex-col w-full"
            >
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(198, 205, 101);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  Test
                </p>
              </button>
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(76, 148, 200);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  ANother
                </p>
              </button>
            </div>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              17
            </span>
            <div
              class="flex flex-col w-full"
            >
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(76, 148, 200);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  ANother
                </p>
              </button>
            </div>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              18
            </span>
            <div
              class="flex flex-col w-full"
            >
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(76, 148, 200);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  ANother
                </p>
              </button>
            </div>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              19
            </span>
            <div
              class="flex flex-col w-full"
            >
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(76, 148, 200);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  ANother
                </p>
              </button>
            </div>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              20
            </span>
            <div
              class="flex flex-col w-full"
            >
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(76, 148, 200);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  ANother
                </p>
              </button>
            </div>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              21
            </span>
            <div
              class="flex flex-col w-full"
            >
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(76, 148, 200);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  ANother
                </p>
              </button>
            </div>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              22
            </span>
            <div
              class="flex flex-col w-full"
            >
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(76, 148, 200);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  ANother
                </p>
              </button>
            </div>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              23
            </span>
            <div
              class="flex flex-col w-full"
            >
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(76, 148, 200);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  ANother
                </p>
              </button>
            </div>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              24
            </span>
            <div
              class="flex flex-col w-full"
            >
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(76, 148, 200);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  ANother
                </p>
              </button>
            </div>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              25
            </span>
            <div
              class="flex flex-col w-full"
            >
              <button
                class="flex-1 px-1 overflow-hidden"
                style="background: rgb(76, 148, 200);"
              >
                <p
                  class="text-white text-sm truncate"
                >
                  ANother
                </p>
              </button>
            </div>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              26
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              27
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              28
            </span>
          </div>
          <div
            class="min-h-32 bg-gray-100 p-3"
          >
            <span
              class="text-lg"
            >
              29
            </span>
          </div>
          <div
            class="h-full bg-gray-300"
          />
          <div
            class="h-full bg-gray-300"
          />
          <div
            class="h-full bg-gray-300"
          />
          <div
            class="h-full bg-gray-300"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
